File: basic/combine functions into transform/javascript/example2.js
/*
Contexto: Tienes una lista de números y deseas filtrar solo los números pares de la lista.
Actualmente, tienes una función que verifica si un número es par y otra función que aplica esta verificación a todos los números en la lista.
Debes combinar estas dos funciones en una sola función de transformación.
*/
// Función para verificar si un número es par
const esPar = (numero) => numero % 2 === 0;

// Función para filtrar los números pares en una lista
const filtrarNumerosPares = (listaNumeros) => {
  const numerosPares = listaNumeros.filter((numero) => esPar(numero));
  return numerosPares;
};

class Int {
    static pairsFrom(nums) {
        const pairs = nums.filter((num) => num % 2 == 0);
        return pairs;
    }
}

module.exports = {filtrarNumerosPares, Int};

File: basic/combine functions into transform/python/src/example_2.py
"""
Contexto: Tienes un texto en formato no normalizado con letras mayúsculas y minúsculas mixtas.
Debes normalizar el texto convirtiendo todas las letras a minúsculas.
Actualmente, tienes una función que convierte una letra a minúscula y otra función que aplica esta conversión a todo el texto.
Debes combinar estas dos funciones en una sola función de transformación.
"""
# Función para convertir una letra a minúscula
def convertir_a_minuscula(letra):
    return letra.lower()

# Función para normalizar un texto completo
def normalizar_texto(texto):
    texto_normalizado = ""
    for letra in texto:
        texto_normalizado += convertir_a_minuscula(letra)
    return texto_normalizado

class String:
    @staticmethod
    def normalize_to_lowercase(text: str):
        return text.lower()